packages/langium/src/workspace/configuration.ts

```diff
@@ -0,0 +1,94 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { DidChangeConfigurationParams } from 'vscode-languageserver';
+import { ConfigurationItem, DidChangeConfigurationNotification } from 'vscode-languageserver-protocol';
+import { LangiumSharedServices } from '../services';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export interface ConfigurationProvider {
+    /**
+    * Returns a configuration value stored for the given language.
+    *
+    * @language the language id
+    * @configuration configuration name
+    */
+    getConfiguration(language: string, configuration: string): Promise<any>;
+
+    /**
+     *  Updates the cached configurations using the `change` notification parameters.
+     *
+     * @param change The parameters of a change configuration notification.
+     * `settings` property of the change object could be expressed as `Record<string, Record<string, any>>`
+     */
+    updateConfiguration(change: DidChangeConfigurationParams): void;
+}
+
+export class DefaultConfigurationProvider implements ConfigurationProvider {
+
+    protected settings: Record<string, Record<string, any>> = {};
+    protected workspaceConfig = false;
+    protected initialized = false;
+    protected readonly sharedServices: LangiumSharedServices;
```

Comment:
I think it's better to store only the actual dependencies (`ServiceRegistry` and `lsp.Connection`) instead of the whole DI container as a class field. This makes it easier to keep track of the service dependencies.

Reply:
Changed. Thanks!